src/MongoDB.Driver/Linq/Linq3Implementation/Translators/ExpressionToAggregationExpressionTranslators/NewExpressionToAggregationExpressionTranslator.cs

```diff
@@ -20,7 +20,6 @@
 using MongoDB.Bson.Serialization;
 using MongoDB.Driver.Linq.Linq3Implementation.Ast;
 using MongoDB.Driver.Linq.Linq3Implementation.Ast.Expressions;
-using MongoDB.Driver.Linq.Linq3Implementation.Misc;
```

Comment:
The problem that I ran into here is that you're building up `computedFields` and calling `ExpressionToAggregationExpressionTranslator` to translate the fields while creating the `ClassMap`. If you've got ideas of how to do this while leveraging the `KnownSerializersFinder` then I'm all ears.

Reply:
Perhaps leave a comment.

But the current implementation has two problems:

1. We're not using the known serializer here, we're creating a new one
2. The known serializer is wrong because it used `AutoMap` and ignored any known child property serializers